src/shims/linux_stubs.h

```diff
@@ -71,7 +71,9 @@ typedef void (*dispatch_mach_msg_destructor_t)(void*);
 #endif
 
 // SIZE_T_MAX should not be hardcoded like this here.
+#ifndef SIZE_T_MAX
 #define SIZE_T_MAX (0x7fffffff)
```

Comment:
while you're at it, can you change this as `(~(size_t)0)` please, this just makes me cringe each time I see it :P